reframe/core/pipeline.py

```diff
@@ -652,7 +653,23 @@ def pipeline_hooks(cls):
     #:       - The old syntax using a ``(h, m, s)`` tuple is dropped.
     #:       - Support of `timedelta` objects is dropped.
     #:       - Number values are now accepted.
-    time_limit = variable(type(None), field=fields.TimerField, value='10m')
+    #:
+    #:    .. versionchanged:: 3.6
+    #:       - The default value is now :class:`None`.
+    #:       - Can be now set per partition via the configuration.
+    time_limit = variable(type(None), field=fields.TimerField, value=None)
+
+    #: .. versionadded:: 3.6
+    #:
+    #: The time limit for the build phase of the regression test.
+    #:
+    #: It is specified similarly to the :attr:`time_limit` attribute.
+    #:
+    #: :type: :class:`str` or :class:`float` or :class:`int`
+    #: :default: ``None``
+    build_time_limit = variable(type(None), field=fields.TimerField,
```

Comment:
I would suggest, but we can vote on it.

```suggestion
    time_limit_build = variable(type(None), field=fields.TimerField,
```